plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/SortFieldUtils.java

```diff
@@ -72,6 +73,26 @@ public static SortOrder parseSortFields(Schema schema, List<String> fields)
         return sortOrder;
     }
 
+    private static Set<Integer> collectColumnFieldIds(Schema schema)
+    {
+        Set<Integer> ids = new HashSet<Integer>();
+        schema.columns().forEach(column -> addNestedField(ids, column));
+        return ids;
+    }
+
+    private static void addNestedField(Set<Integer> baseColumnFieldIds, NestedField field)
+    {
+        baseColumnFieldIds.add(field.fieldId());
+
+        Type type = field.type();
+        if (type.isNestedType()) {
+            NestedType nestedType = type.asNestedType();
+            for (NestedField nestedField : nestedType.fields()) {
+                addNestedField(baseColumnFieldIds, nestedField);
+            }
+        }
+    }
```

Comment:
you can do this with `com.google.common.graph.Traverser`, without need for helper method.